src/dom.js

```diff
@@ -0,0 +1,387 @@
+/* eslint-disable consistent-return */
+import { projectFactory, deleteProject } from './modules/project';
+import { todoFactory, deleteTodo, completeTodo } from './modules/todo';
+
+const EXISTING_DATA = JSON.parse(localStorage.getItem('Projects')) || [];
```

Comment:
As your EXISTING_DATA instance is a const, Js will treat it as an immutable instance, this will probably throw an error when you will try to do things like `EXISTING_DATA.push(newProject)`